### PR TITLE
Support double-publishing of both ESM and CJS modules

### LINKED ISSUE #65: Publish an ESM version to npm
Related to https://github.com/storybookjs/storybook/issues/14910

TeleJSON breaks with Vite 2.3.x (Used in storybook-builder-vite) because TeleJSON is only published to npm as a CJS module. Would it be possible to double-publish this library - like Storybook - so that its source code is TypeScript, but transpiled and published to npm both as CJS and ESM?

### PR DESCRIPTION
**Pull request by eirslett**
Closes #65

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,8 +26,16 @@
   "license": "MIT",
   "main": "dist/index.js",
   "types": "dist/index.d.ts",
+  "module": "dist/index.mjs",
+  "exports": {
+    "import": "./dist/index.mjs",
+    "require": "./dist/index.js"
+  },
```
Comment (ndelangen) [inline package.json:33]: Never seen this before, got any documentation how this works, what it's for?

**Review by ndelangen — APPROVED** [review]

**Comment by ndelangen** [comment]
@eirslett can you help me?

Seems this broke the setup here:
https://app.circleci.com/pipelines/github/storybookjs/storybook/18499/workflows/6194882c-3149-4f83-8074-4cc4eb7bd279/jobs/318298

**Comment by kurtinatlanta** [comment]
You may need to put the cjs and esm files in separate directories so that they don't confuse the builder. Just a guess, though.

File: package.json (hunk 1 above)
Comment (SimenB) [reply package.json:33]: https://nodejs.org/api/packages.html#packages_exports

**Comment by eirslett** [comment]
@ndelangen oops, I didn't consider that TypeScript imports files without a file extension, which is why it's confused when all the files are in the same folder 🤦  Like @kurtinatlanta mentions, the solution is to do like in Storybook - CJS into a CJS folder, ESM into an ESM folder, and then types - into a "types" folder maybe, or just in the root directory?

**Pull request merged**